src/vs/base/browser/ui/grid/grid.ts

```diff
@@ -644,7 +644,7 @@ export class Grid<T extends IView = IView> extends Disposable {
 	}
 
 	private getViewLocation(view: T): GridLocation {
-		const element = this.views.get(view);
+		const element = this.views.get(view) ?? [...this.views.values()][0];
```

Comment:
What is the purpose of this change?

Reply:
These are hacks and should not be shipped as is. I encountered issues with the lifecycle of the view attached to the active GridWidget. It seems that when we close all editors or [dispose this.gridwidget](https://github.com/microsoft/vscode/blob/b782ae6e67c92464bf0306b9466f36083fc2a403/src/vs/workbench/browser/parts/editor/editorPart.ts#L1119-L1120), we still keep around one grid widget (and therefore one view, breadcrumb control etc.) which never gets disposed. This presents issues when trying to restore serialized state, because when deserializing editor view nodes, the previous view and breadcrumb control are still lying around, so getting the active view location and setting the breadcrumb control will throw. I am not familiar enough with the gridview to understand why this is desirable behavior and would appreciate a pointer on how to avoid this hack.

Reply:
You would need to discuss with Grid owner @joaomoreno

Reply:
@bpasero The grid's [`IView` interface](https://github.com/microsoft/vscode/blob/9d0b7ce7306f0cbf9e744cea9221d71277cbd344/src/vs/base/browser/ui/grid/grid.ts#L37-L38) isn't disposable. The grid doesn't own each view's lifecycle. It won't dispose them, otherwise it would also have to conceptually create them. Since it's up to the grid user to create the views (which from the grid's viewpoint aren't "alive"), it's also up to the grid user to dispose them, if needed.

The grid is only disposable because it listens to DOM events, eg. mouse click on the sashes.